webpages/settings/index.js

```diff
@@ -35,18 +35,17 @@ const ColorInput = Vue.extend({
     };
   },
   ready() {
-    self = this;
+    let that = this
     if (this.no_alpha === "true") {
       this.formats = "hex,rgb,hsv,hsl";
     } else {
       this.formats = "hex,hex8,rgb,hsv,hsl";
     }
-    console.log(this.no_alpha);
-    this.$els.pickr.addEventListener("change", (e) => {
-      self.color = "#" + e.detail.value;
-      if (self.value !== self.color) {
-        self.$parent.addonSettings[self.addon._addonId][self.setting.id] = "#" + self.$els.pickr.hex8;
-        self.$parent.updateSettings(self.addon, { wait: 250, settingId: self.setting.id });
+    this.$els.pickr.addEventListener('change', (e) => {
+      that.color = "#" + e.detail.value;
```

Comment:
You're using an arrow function, so `this === that // true` here, No need to use "that"